src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -167,14 +168,27 @@ default ClassDesc arrayType() {
      *
      * @param rank the rank of the array
      * @return a {@linkplain ClassDesc} describing the array type
-     * @throws IllegalArgumentException if the rank is less than or equal to zero or if the rank of the resulting array type is
+     * @throws IllegalArgumentException if the rank is less than or
+     * equal to zero or if the rank of the resulting array type is
      * greater than 255
      * @jvms 4.4.1 The CONSTANT_Class_info Structure
      */
     default ClassDesc arrayType(int rank) {
-        int currentDepth = ConstantUtils.arrayDepth(descriptorString());
-        if (rank <= 0 || currentDepth + rank > ConstantUtils.MAX_ARRAY_TYPE_DESC_DIMENSIONS)
-            throw new IllegalArgumentException("rank: " + currentDepth + rank);
+        int netRank;
+        if (rank <= 0) {
+            throw new IllegalArgumentException("rank " + rank + "is not a positive value");
```

Comment:
There's a space missing in the IAE message after the rank number. For example, if rank == 0, the message would be: "rank 0is not positive value" ...

Reply:
Good catch; will fix before pushing.